File: src/main/java/_712/final_project_712/model/Product.java
package _712.final_project_712.model;

import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.Table;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Table("product")
public class Product {
    @Id
    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    private Integer stock;
    private Long categoryId;
    private String brand;
    private String model;
    private String specifications;
    private String imageUrl;
    private Integer status;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
} 

File: ui/src/api/product.ts
import request from '@/utils/request'

export interface Product {
  id: number
  name: string
  description: string
  price: number
  stock: number
  categoryId: number
  categoryName: string
  brand: string
  model: string
  specifications: string
  imageUrl: string
  status: number
  createTime: string
  updateTime: string
}

export function getProducts() {
  return request({
    url: '/product/list',
    method: 'get'
  })
}

/** 
 * 获取商品详情
 * @param {number} id 商品ID
 * @returns {Promise<Result<Product>>} 商品详情
 */
export function getProductDetail(id: number) {
  return request({
    url: `/product/${id}`,
    method: 'get'
  })
}

/**
 * 获取所有商品分类
 * @returns {Promise<ProductCategoryListResponse>} 商品分类列表
 */
export function getProductCategories() {
  return request({
    url: '/product/categories',
    method: 'get'
  })
}

/**
 * 获取分类商品列表
 * @param {number} categoryId 分类ID
 * @returns {Promise<Result<Product[]>>} 商品列表
 */
export function getProductsByCategory(categoryId: number) {
  return request({
    url: `/product/category/${categoryId}`,
    method: 'get'
  })
} 